core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -50,10 +50,11 @@ import org.apache.spark.util.{Clock, SystemClock, Utils}
 private[scheduler] class BlacklistTracker (
     private val listenerBus: LiveListenerBus,
     conf: SparkConf,
+    sc: Option[SparkContext],
```

Comment:
+1. I'd rather avoid making changes to `SparkContext` at all. Especially since this particular API doesn't seem particularly useful for an app developer.